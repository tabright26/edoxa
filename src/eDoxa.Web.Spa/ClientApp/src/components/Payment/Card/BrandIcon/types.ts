import { SizeProp } from "@fortawesome/fontawesome-svg-core";

export type BrandProp = "visa" | "amex" | "mastercard" | "discover";

export interface CardBrandIconProps {
  className: string;
  brand: BrandProp;
  size?: SizeProp;
}
