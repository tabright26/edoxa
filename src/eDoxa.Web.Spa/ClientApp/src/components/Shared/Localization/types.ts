export interface Country {
  readonly name: string;
  readonly twoDigitIso: string;
}

export interface Address {
  country: Country;
  line1: string;
  line2: string | null;
  city: string;
  state: string | null;
  postalCode: string;
}
