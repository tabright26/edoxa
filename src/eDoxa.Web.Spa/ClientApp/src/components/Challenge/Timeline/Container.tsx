import { connect, MapStateToProps } from "react-redux";
import Timeline from "./Timeline";
import { RootState } from "store/types";
import { RouteComponentProps, withRouter } from "react-router-dom";
import { ChallengeId, ChallengeTimeline } from "types";
import { compose } from "recompose";

interface Params {
  readonly challengeId: ChallengeId;
}

type OwnProps = RouteComponentProps<Params>;

interface StateProps {
  readonly state: string;
  readonly timeline: ChallengeTimeline;
}

const mapStateToProps: MapStateToProps<StateProps, OwnProps, RootState> = (state, ownProps) => {
  const { data } = state.root.challenge;
  const challenge = data.find(challenge => challenge.id === ownProps.match.params.challengeId);
  return {
    state: challenge.state,
    timeline: challenge.timeline
  };
};

const enhance = compose<any, any>(
  withRouter,
  connect(mapStateToProps)
);

export default enhance(Timeline);
