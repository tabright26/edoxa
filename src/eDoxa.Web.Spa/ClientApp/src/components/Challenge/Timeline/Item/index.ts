import ChallengeTimelineItem from "./Item";

export default ChallengeTimelineItem;
