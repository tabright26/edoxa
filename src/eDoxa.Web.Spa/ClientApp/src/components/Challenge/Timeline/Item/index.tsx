import React, { FunctionComponent } from "react";
import Moment from "react-moment";

type Props = {
  readonly currentState: string;
  readonly state: string;
  readonly unixTimeSeconds: number;
};

const Item: FunctionComponent<Props> = ({
  currentState,
  state,
  unixTimeSeconds
}) => {
  const isActiveState = currentState.toLowerCase() === state.toLowerCase();
  return (
    <span
      className={`btn btn-block btn-sm text-light mt-2 ${
        isActiveState ? "bg-primary" : "bg-secondary"
      }`}
      title={`${state}${isActiveState ? " (current)" : ""}`}
    >
      {unixTimeSeconds ? (
        <Moment unix format="lll">
          {unixTimeSeconds}
        </Moment>
      ) : (
        <span>...</span>
      )}
    </span>
  );
};

export default Item;
