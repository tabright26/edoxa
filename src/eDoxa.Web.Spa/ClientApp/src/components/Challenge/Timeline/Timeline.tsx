import React, { FunctionComponent } from "react";
import ChallengeTimelineItem from "./Item";
import { ChallengeTimeline } from "types";

const style = { width: "150px" };

interface Props {
  readonly state: string;
  readonly timeline: ChallengeTimeline;
}

const Timeline: FunctionComponent<Props> = ({ state, timeline }) => (
  <div
    className="d-flex flex-column position-relative"
    style={{
      right: "50px"
    }}
  >
    <span className="btn bg-gray-900 mt-2 btn-sm rounded-0" style={style}>
      <strong className="text-uppercase">Timeline</strong>
    </span>
    <ChallengeTimelineItem currentState={state} state="Inscription" unixTimeSeconds={timeline.createdAt} />
    <ChallengeTimelineItem currentState={state} state="Started" unixTimeSeconds={timeline.startedAt} />
    <ChallengeTimelineItem currentState={state} state="Ended" unixTimeSeconds={timeline.endedAt} />
    <ChallengeTimelineItem currentState={state} state="Closed" unixTimeSeconds={timeline.closedAt} />
  </div>
);

export default Timeline;
