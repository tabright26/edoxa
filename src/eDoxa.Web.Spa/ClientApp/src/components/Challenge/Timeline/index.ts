import ChallengeScoring from "./Container";

export default ChallengeScoring;
