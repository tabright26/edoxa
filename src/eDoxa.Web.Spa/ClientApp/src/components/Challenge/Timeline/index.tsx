import React, { FunctionComponent } from "react";
import Item from "./Item";
import { connect, MapStateToProps } from "react-redux";
import { RootState } from "store/types";
import { RouteComponentProps, withRouter } from "react-router-dom";
import { ChallengeId, ChallengeTimeline } from "types";
import { compose } from "recompose";

type Params = {
  readonly challengeId: ChallengeId;
};

type OwnProps = RouteComponentProps<Params>;

type StateProps = {
  readonly state: string;
  readonly timeline: ChallengeTimeline;
};

type InnerProps = OwnProps & StateProps;

type OutterProps = {};

type Props = InnerProps & OutterProps;

const Timeline: FunctionComponent<Props> = ({ state, timeline }) => (
  <div
    className="d-flex flex-column position-relative"
    style={{
      right: "50px"
    }}
  >
    <span
      className="btn bg-gray-900 mt-2 btn-sm rounded-0"
      style={{ width: "150px" }}
    >
      <strong className="text-uppercase">Timeline</strong>
    </span>
    <Item
      currentState={state}
      state="Inscription"
      unixTimeSeconds={timeline.createdAt}
    />
    <Item
      currentState={state}
      state="Started"
      unixTimeSeconds={timeline.startedAt}
    />
    <Item
      currentState={state}
      state="Ended"
      unixTimeSeconds={timeline.endedAt}
    />
    <Item
      currentState={state}
      state="Closed"
      unixTimeSeconds={timeline.closedAt}
    />
  </div>
);

const mapStateToProps: MapStateToProps<StateProps, OwnProps, RootState> = (
  state,
  ownProps
) => {
  const { data } = state.root.challenge;
  const challenge = data.find(
    challenge => challenge.id === ownProps.match.params.challengeId
  );
  return {
    state: challenge.state,
    timeline: challenge.timeline
  };
};

const enhance = compose<InnerProps, OutterProps>(
  withRouter,
  connect(mapStateToProps)
);

export default enhance(Timeline);
