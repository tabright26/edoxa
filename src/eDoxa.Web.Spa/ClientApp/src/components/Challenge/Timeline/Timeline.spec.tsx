import React from "react";
import renderer from "react-test-renderer";
import Timeline from "./Timeline";
import { ChallengeTimeline } from "types";

it("renders without crashing", () => {
  // Arrange

  const state: string = "";
  const timeline: ChallengeTimeline = {
    createdAt: Date.now(),
    startedAt: null,
    endedAt: null,
    closedAt: null
  };

  // Act
  const tree = renderer.create(<Timeline state={state} timeline={timeline} />).toJSON();

  // Assert
  expect(tree).toMatchSnapshot();
});
