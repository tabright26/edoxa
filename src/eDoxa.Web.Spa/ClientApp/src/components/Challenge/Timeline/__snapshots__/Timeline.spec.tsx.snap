// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="d-flex flex-column position-relative"
  style={
    Object {
      "right": "50px",
    }
  }
>
  <span
    className="btn bg-gray-900 mt-2 btn-sm rounded-0"
    style={
      Object {
        "width": "150px",
      }
    }
  >
    <strong
      className="text-uppercase"
    >
      Timeline
    </strong>
  </span>
  <span
    className="btn bg-secondary btn-sm text-light mt-2 rounded-0"
    style={
      Object {
        "width": "150px",
      }
    }
    title="Inscription"
  >
    <time
      dateTime={"+051830-07-11T06:22:48.000Z"}
    >
      Jul 11, 51830 2:22 AM
    </time>
  </span>
  <span
    className="btn bg-secondary btn-sm text-light mt-2 rounded-0"
    style={
      Object {
        "width": "150px",
      }
    }
    title="Started"
  >
    <span>
      ...
    </span>
  </span>
  <span
    className="btn bg-secondary btn-sm text-light mt-2 rounded-0"
    style={
      Object {
        "width": "150px",
      }
    }
    title="Ended"
  >
    <span>
      ...
    </span>
  </span>
  <span
    className="btn bg-secondary btn-sm text-light mt-2 rounded-0"
    style={
      Object {
        "width": "150px",
      }
    }
    title="Closed"
  >
    <span>
      ...
    </span>
  </span>
</div>
`;
